File: 2021/day-01.py
nums = [int(line.strip()) for line in open('input/day-01-input.txt')]

def solve(nums, skip):
    return len([i for i in range(len(nums) - skip) if nums[i] < nums[i+skip]])

print(solve(nums, 1))
print(solve(nums, 3))

def solve_alt(nums, window):
    return sum(a < b for a, b in zip(nums, nums[window:]))

print(solve_alt(nums, 1))
print(solve_alt(nums, 3))

File: 2021/day-01.pl
use strict;
use warnings;
use v5.30;
use Path::Tiny;

my @input = split /\n/, path('input/day-01-input.txt')->slurp;

sub solve {
    my ( $input, $window ) = @_;
    scalar grep { @$input[$_] < @$input[ $_ + $window ] } 0 .. $#$input - $window;
}

say solve \@input, 1;
say solve \@input, 3;
